src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -44,16 +55,51 @@ export class MarkdownHover implements IHoverPart {
 	}
 }
 
+export class VerboseMarkdownHover extends MarkdownHover {
+
+	constructor(
+		owner: IEditorHoverParticipant<MarkdownHover>,
+		range: Range,
+		contents: IMarkdownString[],
+		isBeforeContent: boolean,
+		ordinal: number,
+		public readonly hoverId?: string,
+		public readonly sourceProvider?: HoverProvider,
+		public readonly verbosityMetadata?: HoverVerbosityMetadata,
+		public readonly dispose?: () => void
+	) {
+		super(owner, range, contents, isBeforeContent, ordinal);
+	}
+}
+
+interface HoverVerbosityMetadata {
+	canIncreaseVerbosity?: boolean;
+	canDecreaseVerbosity?: boolean;
+}
+
+interface FocusedHoverInfo {
+	focusedIndex: number;
+	focusedElement: HTMLElement;
```

Comment:
I hope I didn't say this wrong before - because `_focusInfo` already indicates that it describes the focused element, the fields here don't have to re-iterate that.

```suggestion
	hoverIndex: number;
	hoverElement: HTMLElement;
```